gio-subclass/src/auto/action.rs

```diff
@@ -118,14 +118,19 @@ unsafe extern "C" fn action_get_name<T: ObjectType>
     let imp = instance.get_impl();
     let imp = (*(*interface_static).imp_static).get_impl(imp);
     let wrap = from_glib_borrow(gptr);
+
+    unsafe extern "C" fn destroy(p: glib_ffi::gpointer){
+        glib_ffi::g_free(p);
+    };
+
     match imp.get_name(&wrap)
     {
         Some(t) => {
             let ret = t/*Not checked*/.to_glib_none().0;
             gobject_ffi::g_object_set_qdata_full(gptr as *mut gobject_ffi::GObject,
                 glib_ffi::g_quark_from_string("rs_get_name".to_glib_none().0),
                 ret as *mut c_void,
-                Some(<String as FreeGlibPtr>::free)
+                Some(destroy)
```

Comment:
Would you prefer to see it like this?
I guess that works too, but I find that that would end up as a lot of duplicate code. Does the gir know what ```free``` function to call? Like ```get_type```? Because if not, I don't think we can do much different than ```FreeGlibPtr```

Reply:
Apart from `from_glib_full()` and having things go out of scope, I don't see any other solution either. And for strings that involves another copy until I fix that, for other types it should be fine.

So maybe we can go with `from_glib_full()` for now and assume the string problem is solved soon. It's on my list for the next hackfest

Reply:
So I'll go with a nested destroy function that calls ````from_glib_full```.

Reply:
That's what I would do for now, and if there's a performance problem we can always implement that manually until the string problem is solved